core/src/main/java/org/geonetwork/map/wms/SLDUtil.java

```diff
@@ -0,0 +1,248 @@
+     */
+    public static String encodeFilter(Filter filter) throws IOException {
+        OutputStream outputStream = new ByteArrayOutputStream();
+
+        Configuration configuration = new OGCConfiguration();
+        Encoder encoder = new Encoder(configuration);
+        encoder.encode( filter, org.geotools.filter.v1_1.OGC.Filter, outputStream);
+
+        return outputStream.toString();
+    }
+
+    /**
+     * Generate a SLD Filter from filters defined in a JSON
+     *
+     * JSON example :
+     * <pre>
+     * {
+     *     "baseStyle" : "Test:MuiltiRoad",
+     *     "filters": [ {"field_name": "longueur",
+     *                   "filter": [ { "filter_type": "PropertyIsBetween",
+     *                                 "params": [0,500]},
+     *                               { "filter_type": "PropertyIsBetween",
+     *                                 "params": [500,5000]}]},
+     *                  {"field_name": "departement",
+     *                   "filter": [ { "filter_type": "PropertyIsEqualTo",
+     *                                 "params": ["Ain"]}]},
+     *                  {"field_name": "date_renovation",
+     *                   "filter": [ { "filter_type": "PropertyIsBetween",
+     *                                 "params": ["2015-07-01", "2015-08-31"]},
+     *                               { "filter_type": "PropertyIsBetween",
+     *                                 "params": ["2014-07-01", "2014-08-31"]}]}
+     *                ]
+     * }</pre>
+     *
+     * @param userFilters JSON representation of filters
+     * @return Filter instance that represent combination of filters specified in JSON
+     * @throws JSONException if some have wrong parameter count or malformed JSON
+     */
+
+    public static Filter generateCustomFilter(JSONObject userFilters) throws JSONException {
+        FilterFactory2 ff2 = CommonFactoryFinder.getFilterFactory2();
+
+        JSONArray filters = userFilters.getJSONArray("filters");
+        List<Filter> res = new LinkedList<Filter>();
+
+        for(int i=0;i<filters.length();i++)
+            res.add(SLDUtil.generateFilter(filters.getJSONObject(i)));
+
+        if(res.size() > 1)
+            return ff2.and(res);
+        else
+            return res.get(0);
+
+    }
+
+    private static Filter generateFilter(JSONObject jsonObject) throws JSONException {
+
+        FilterFactory2 ff2 = CommonFactoryFinder.getFilterFactory2();
+
+        String fieldName = jsonObject.getString("field_name");
+
+        List<Filter> res = new LinkedList<Filter>();
+
+        JSONArray filters = jsonObject.getJSONArray("filter");
+        for(int i=0;i<filters.length();i++)
+            res.add(SLDUtil.generateFilter2(fieldName, filters.getJSONObject(i)));
+
+        if(res.size() > 1)
+            return ff2.or(res);
+        else
+            return  res.get(0);
+    }
+
+    private static Filter generateFilter2(String fieldName, JSONObject jsonObject) throws JSONException {
```

Comment:
The doc is basically the one of the public method https://github.com/geonetwork/core-geonetwork/pull/1568/files/278705d830846bc843b7cc5cc91595a7c1c8f193#diff-92b8f7f0fd2069bcb9128699d3b63bf2R141 add a filter defined in JSON to the SLD.